packages/core/src/transforms/resize.ts

```diff
@@ -24,28 +45,46 @@ export const resize: TransformFactory<ResizeOptions> = (config, ctx) => {
         : config.h
             ? parseInt(config.h)
             : undefined
+    const aspect = width && height
+        ? width / height
+        : parseAspect(config.aspect)
 
-    if (!width && !height) return
+    if (!width && !height && !aspect) return
 
     return function resizeTransform(image) {
+        let finalWidth = width
+        let finalHeight = height
+
+        const w = getMetadata(image, 'width')
+        const h = getMetadata(image, 'height')
 
-        if (!height) {
-            const w = getMetadata(image, 'width')
-            const h = getMetadata(image, 'height')
-            setMetadata(image, 'height', Math.round((width! / w) * h))
-            setMetadata(image, 'width', width)
+        const metaAspect = w / h
```

Comment:
I don't know why you'd need to use the metadata at this point, since -- and correct me if I'm wrong -- it should `getMetadat(image,'width')` should always return `undefined` at this point. Or am I missing something?

Reply:
I actually wasn't sure if this would work at first either, but the metadata is already populated from `sharp`

The case where only `aspect` is given takes advantage of it, looking at the metadata from `sharp` to decide which axis needs to be cropped down

Reply:
Oh duh 🤦‍♂️ yes you're absolutely right